/**
 * Copyright Soramitsu Co., Ltd. All Rights Reserved.
 * SPDX-License-Identifier: Apache-2.0
 */

syntax = "proto3";
package libp2p.crypto.proto;

enum KeyType {
  UNSPECIFIED = 0;
  RSA1024 = 1;
  RSA2048 = 2;
  RSA4096 = 3;
  ED25519 = 4;
  SECP256K1 = 5;
}

message PrivateKey {
  KeyType key_type = 1;
  bytes key_value = 2;
}

message PublicKey {
  KeyType key_type = 1;
  bytes key_value = 2;
}
